### ISSUE TITLE
some DbAdapter driver_options can make DbSelect::count() incorrect

### ISSUE THREAD

**Issue by richcem** [laminas/laminas-paginator#10]
### Bug Report

some DbAdapter driver_options can make DbSelect::count() incorrect

|    Q        |   A
|------------ | ------
| Version(s)  | 2.8.2

#### Summary

PDO::CASE_LOWER can make DbSelect::count() incorrect.
#### Current behavior

get NOTICE: Undefined index: C on xxxxx

#### How to reproduce

```php
$adapter = new Adapter([
            'driver'   => 'pdo_mysql',
            "driver_options" => [
                \PDO::ATTR_CASE => \PDO::CASE_LOWER,
            ],
         // other options ....
]);

$select = $sql->select();

$dbSelect = new DbSelect($select, xxxx)
$dbSelect->count();

// NOTICE: Undefined index: C on xxxxx
```

```php
// DbSelect::count()
public function count()
    {
        if ($this->rowCount !== null) {
            return $this->rowCount;
        }

        $select = $this->getSelectCount();

        $statement = $this->sql->prepareStatementForSqlObject($select);
        $result    = $statement->execute();
        $row       = $result->current();

        // !!! here return $row['c'] = 10，lower case key

        $this->rowCount = (int) $row[self::ROW_COUNT_COLUMN_NAME];

        return $this->rowCount;
    }
```

#### Expected behavior

check lower or upper case key，return the right count

**Comment by samsonasik** [comment]
I created PR https://github.com/laminas/laminas-paginator/pull/11 for it. Haven't added unit test yet ...

**Comment by weierophinney** [comment]
Fixed with #11.
